### PR TITLE
Add PublishBlob UnitTest

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

We want to test the following two scenarios:
1) Client A makes a block proposal with a blob. One validator is down. The other three validated the block but don’t confirm it. Now the validator comes back up but another one goes down. Client B (who doesn’t have the blob) then re-proposes that block in the next round.
2) Client A proposes a block with a blob, and it gets confirmed, but one of the four validators is down and doesn’t learn about it at all. Then that validator comes back up but another goes down (so now only two are up and have the blob). Client B proposes a child of that block.

## Proposal

For 1), we modified the existing `test_propose_validated` test.
For 2), we created a new test `test_publish_blob`.

## Test Plan

The tests created

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#2062** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2062?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-core/src/unit_tests/client_tests.rs [hunk 1]
```diff
@@ -1599,8 +1682,23 @@ where
...
+        Amount::from_tokens(9)
     );
+    assert!(client0.pending_block.is_none());
+
+    // Burn another token so Client 1 sees that the blob is already published
+    client1
+        .burn(None, Amount::from_tokens(1), UserData::default())
+        .await
+        .unwrap();
+    client1.synchronize_from_validators().await.unwrap();
```
Comment (afck) [inline linera-core/src/unit_tests/client_tests.rs:1694]: Shouldn't this be enough to see that it's published? No need to burn another precious token!

**Review by afck — APPROVED** [review]
Looks great, thanks! :+1:

File: linera-core/src/unit_tests/client_tests.rs (hunk 1 above)
Comment (andresilva91) [reply linera-core/src/unit_tests/client_tests.rs:1694]: Synchronizing should be enough, but I wanted to make sure by checking the pending block/blobs, and that'll only get cleared if I do another operation :/

**Pull request merged**